solr/core/src/test/org/apache/solr/core/TestFileSystemConfigSetService.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.solr.core;
+
+import static org.apache.solr.core.FileSystemConfigSetService.METADATA_FILE;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Map;
+import org.apache.commons.io.file.PathUtils;
+import org.apache.solr.SolrTestCaseJ4;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class TestFileSystemConfigSetService extends SolrTestCaseJ4 {
+  private static Path configSetBase;
+  private static FileSystemConfigSetService fileSystemConfigSetService;
+
+  @BeforeClass
+  public static void beforeClass() throws Exception {
+    configSetBase = createTempDir();
+    fileSystemConfigSetService = new FileSystemConfigSetService(configSetBase);
+  }
+
+  @AfterClass
+  public static void afterClass() throws Exception {
+    PathUtils.deleteDirectory(configSetBase);
+    fileSystemConfigSetService = null;
+  }
+
+  @Test
+  public void testUploadConfig() throws IOException {
```

Comment:
I checked with "/" input as a configName and  there is access denied FilePermission exception.

Reply:
That sounds like you were stopped by the Java security manager or OS.  If you change the path to someplace the test can write to then I think it will show the flaw

Reply:
Yes, it is java security manager. I checked relative configName path to configSetBase and throw IOException.

Reply:
When it comes to FileSystemConfigSetService.locateInstanceDir, in case the presence of configset, we still need the dir for the configset.